src/librustc_middle/ty/relate.rs

```diff
@@ -617,12 +617,22 @@ impl<'tcx> Relate<'tcx> for &'tcx ty::List<ty::ExistentialPredicate<'tcx>> {
         a: &Self,
         b: &Self,
     ) -> RelateResult<'tcx, Self> {
-        if a.len() != b.len() {
+        let tcx = relation.tcx();
+
+        // FIXME: this is wasteful, but want to do a perf run to see how slow it is.
```

Comment:
I wonder if we should open up a FIXME issue for this..? To try and prevent the duplicates in the first place and figure out where they come from?